### DISCUSSION TITLE
What's best way to migrate protocol to libp2p from existing protocol

### DISCUSSION THREAD

**Discussion by canewsin** [libp2p/rust-libp2p#4214] [Q&A]
I tried libp2p previously, where both were libp2p clients. currently i was rewriting ZeroNet into rust @https://github.com/ZeroNetX/zeronet-rs from Python, instead of rewriting network layer, we want to migrate protocol into libp2p, afaik libp2p communication uses noise to encrypt communication and connection couldn't be made with PlainText1Config when I last tried.

So my question is, is it possible to create a communication layer between a libp2p and non-libp2p client ?

**Comment by thomaseizinger** [comment]
It depends on what your protocol looks like on the wire.

To interoperate successfully, a node needs to follow certain specifications, see https://github.com/libp2p/specs for details.

For example, for a QUIC connection, we require a particular TLS extension. For a TCP connection, we expect the use of `multistream-select` to negotiate the use of a security protocol like noise and a muxer like yamux.

There is work under way to also support running on top of HTTP transports but that is still in the design phase, see https://github.com/libp2p/specs/pull/508.

**Comment by canewsin** [comment]
@thomaseizinger thanks for the reply.
found some info here as well 
https://discuss.libp2p.io/t/is-libp2p-suitable-to-build-a-node-for-a-existing-non-libp2p-network/1769
